volttron_data/agent_templates/common/agent.py_

```diff
@@ -0,0 +1,145 @@
+"""
+Agent documentation goes here.
+"""
+
+__docformat__ = 'reStructuredText'
+
+import logging
+import sys
+from volttron.platform.agent import utils
+from volttron.platform.vip.agent import Agent, Core, RPC
+
+_log = logging.getLogger(__name__)
+utils.setup_logging()
+__version__ = "__version_string__"
+
+
+def __module_name__(config_path, **kwargs):
+    """Parses the Agent configuration and returns an instance of
+    the agent created using that configuration.
+
+    :param config_path: Path to a configuration file.
+
+    :type config_path: str
+    :returns: __class_name__
+    :rtype: __class_name__
+    """
+    try:
+        config = utils.load_config(config_path)
+    except StandardError:
+        config = {}
+
+    if not config:
+        _log.info("Using Agent defaults for starting configuration.")
+
+    setting1 = int(config.get('setting1', 1))
+    setting2 = config.get('setting2', "some/random/topic")
+
+    return __class_name__(setting1,
```

Comment:
Were you going to add the other settings in here as well? Or no, either way is fine with me.